velox/common/memory/MemoryArbitrator.h

```diff
@@ -86,12 +71,29 @@ class MemoryArbitrator {
     /// happens to trigger the failed memory arbitration.
     bool retryArbitrationFailure{true};
   };
+
+  using Factory = std::function<std::unique_ptr<MemoryArbitrator>(
+      const MemoryArbitrator::Config& config)>;
+
+  /// Register factory for a specific kind of memory arbitrator.
+  /// MemoryArbitrator::Create looks up the registry to find the appropriate
+  /// factory to create arbitrator instance using the kind string in arbitrator
+  /// configuration.
+  static void registerFactory(const std::string& kind, Factory factory);
+
+  /// Invoked by the memory manager to create an instance of memory arbitrator
```

Comment:
```
/// Invoked by the memory manager to create an instance of memory arbitrator based on the kind specified in 'config'. The arbitrator kind must be registered through MemoryArbitrator::registerFactory, otherwise the function throws a velox user exception error.
///
/// NOTE: if arbitrator kind is not set in 'config', then the function returns nullptr.
```